tools/codegen/gofilegen/resource/resource_file.go

```
@@ -17,7 +17,7 @@ func GenerateGoCode(input *codespec.Resource) ([]byte, error) {
 		APIOperations: codetemplate.APIOperations{
 			VersionHeader: input.Operations.VersionHeader,
 			Create:        *toCodeTemplateOpModel(&input.Operations.Create),
-			Update:        *toCodeTemplateOpModel(&input.Operations.Update),
+			Update:        toCodeTemplateOpModel(input.Operations.Update),
```

probably out-of-scope. But could potentially use force_replace on every single attribute that is not computed

---

Yes, out of scope but I think this would be the way to go. cc @AgustinBettati

---

If I understood correctly, I think it depends :-) 

for resources that can be recreated without loosing data, force_replace is ok, however for resource potentially loosing data (e.g. cluster), it's not ok to create the resource if an attribute (e.g. cluster name or project_id) changes, in that case I think it's better to raise an error that that attribute can't be changed